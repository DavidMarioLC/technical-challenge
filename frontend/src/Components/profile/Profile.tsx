import "./profile.css";
import {TiStarOutline} from 'react-icons/ti';
import {HiOutlineOfficeBuilding} from 'react-icons/hi';
import {FiMapPin,FiUsers} from 'react-icons/fi';
import {BsLink45Deg,BsTwitter} from 'react-icons/bs'

import Avatar from './avatar/Avatar';
import Button from '../common/button/Button';
import Information from './information/Information';
import InformationLink from './information/informationLink/InformationLink';
import Contact from './contacts/Contact'
import ContactLink from './contacts/contactLink/ContactLink';

import {IUser} from '../../interfaces/user.interface';

interface Props{
  user:IUser
}


const Profile = ({user}:Props) => {
  
  const {
    avatar_url,name,
    login:nickname,
    bio,
    followers,
    following,
    company,
    location,
    blog,
    twitter_username} = user;

  return (
    <div className="profile">
    <div className="profile__user">
      <Avatar  avatar_url={avatar_url} name={name} nickname={nickname}/>
    </div>
    <div className="profile__buttons">
          <Button size='large' value='Edit profile'/>  
    </div>
    <div className="profile__details">
      <p className="biography">{bio}</p>
        <Information>
          <InformationLink text='Followers' value={followers} icon={FiUsers}/>
          <InformationLink text='Following' value={following} />
          <InformationLink text='Stars' value={55} icon={TiStarOutline}/>
        </Information>
    </div>
    <div className="profile__contact">
      <Contact>
    	  {company &&  <ContactLink icon={HiOutlineOfficeBuilding} value={company} link={false}/>}
        {location && <ContactLink icon={FiMapPin} value={location} link={false}/>}
        {blog &&  <ContactLink icon={BsLink45Deg} value={blog}/>}
        {twitter_username && <ContactLink icon={BsTwitter} value={twitter_username} link={false}/>}
      </Contact>
    </div>
  </div>
  )
}

export default Profile
