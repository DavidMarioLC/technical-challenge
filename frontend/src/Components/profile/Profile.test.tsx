import renderer from 'react-test-renderer';
import Profile from './Profile';


it('render Profile correctly', () => {
  const tree = renderer.create(<Profile  user={{
    avatar_url: '',
    bio: '',
    blog: '',
    company: '',
    created_at: new Date(),
    email: null,
    events_url: '',
    followers: 0,
    followers_url: '',
    following: 0,
    following_url: '',
    gists_url: '',
    gravatar_id: '',
    hireable: null,
    html_url: '',
    id: 0,
    location: '',
    login: '',
    name: '',
    node_id: '',
    organizations_url: '',
    public_gists: 0,
    public_repos: 0,
    received_events_url: '',
    repos_url: '',
    site_admin: false,
    starred_url: '',
    subscriptions_url: '',
    twitter_username: '',
    type: '',
    updated_at:new Date(),
    url: '',
    }}/>).toJSON();
  expect(tree).toMatchSnapshot();
});