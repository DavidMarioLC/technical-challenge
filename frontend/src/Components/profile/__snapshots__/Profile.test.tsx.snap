// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Profile correctly 1`] = `
<div
  className="profile"
>
  <div
    className="profile__user"
  >
    <div
      className="avatar"
    >
      <figure
        className="avatar__image"
      >
        <img
          alt="username"
          height="80"
          loading="lazy"
          src=""
          title="username"
          width="80"
        />
      </figure>
      <div
        className="avatar__info"
      >
        <p
          className="avatar__fullname"
        >
          
        </p>
        <p
          className="avatar__username"
        >
          
        </p>
      </div>
    </div>
  </div>
  <div
    className="profile__buttons"
  >
    <button
      className="btn btn--large undefined"
    >
      Edit profile
    </button>
  </div>
  <div
    className="profile__details"
  >
    <p
      className="biography"
    >
      
    </p>
    <div
      className="info"
    >
      <a
        className="info__link"
        href="#followers"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
          />
          <circle
            cx="9"
            cy="7"
            r="4"
          />
          <path
            d="M23 21v-2a4 4 0 0 0-3-3.87"
          />
          <path
            d="M16 3.13a4 4 0 0 1 0 7.75"
          />
        </svg>
        <span
          className="info__value"
        >
          0
        </span>
        Followers
      </a>
      <a
        className="info__link"
        href="#followers"
      >
        <span
          className="info__value"
        >
          0
        </span>
        Following
      </a>
      <a
        className="info__link"
        href="#followers"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.855 20.966c-.224 0-.443-.05-.646-.146l-.104-.051-4.107-2.343-4.107 2.344-.106.053c-.488.228-1.085.174-1.521-.143-.469-.34-.701-.933-.586-1.509l.957-4.642-1.602-1.457-1.895-1.725-.078-.082c-.375-.396-.509-.97-.34-1.492.173-.524.62-.912 1.16-1.009l.102-.018 4.701-.521 1.946-4.31.06-.11c.262-.473.764-.771 1.309-.771.543 0 1.044.298 1.309.77l.06.112 1.948 4.312 4.701.521.104.017c.539.1.986.486 1.158 1.012.17.521.035 1.098-.34 1.494l-.078.078-3.498 3.184.957 4.632c.113.587-.118 1.178-.59 1.519-.252.182-.556.281-.874.281zm-8.149-6.564c-.039.182-.466 2.246-.845 4.082l3.643-2.077c.307-.175.684-.175.99 0l3.643 2.075-.849-4.104c-.071-.346.045-.705.308-.942l3.1-2.822-4.168-.461c-.351-.039-.654-.26-.801-.584l-1.728-3.821-1.726 3.821c-.146.322-.45.543-.801.584l-4.168.461 3.1 2.822c.272.246.384.617.302.966z"
          />
        </svg>
        <span
          className="info__value"
        >
          55
        </span>
        Stars
      </a>
    </div>
  </div>
  <div
    className="profile__contact"
  >
    <div>
      
      
      
      
    </div>
  </div>
</div>
`;
